soc/arm/xilinx_zynqmp/Kconfig.defconfig

```diff
@@ -28,6 +28,27 @@ config SYS_CLOCK_HW_CYCLES_PER_SEC
 
 endif # SOC_XILINX_ZYNQMP_RPU
 
+if SOC_XILINX_ZYNQMP_APU
+
+config NUM_IRQS
```

Comment:
I would recommend to get rid of this "NUM_IRQS" configuration and use the Distributor "TYPER" register to read the same information at runtime. Something like this, 

`total_irqs = READ_REGISTER(GIC_TYPER) & 0x1F) + 1) * 32;`

Thoughts ?

Reply:
`NUM_IRQS` can be considered part of "standard SoC definitions" (i.e. all SoCs specify this value, and Zephyr expects it to be specified). Unless we refactor the interrupt number handling, we have to define/use this value.

Moreover, I do not see the advantage of using the run-time number when we can already determine it statically.

Reply:
> Unless we refactor the interrupt number handling, we have to define/use this value.

Its fine to keep it this way..

> Moreover, I do not see the advantage of using the run-time number when we can already > determine it statically.

The only reason I pointed this out is that a single line code addition to Zephyr will make porting new Armv8 SoCs little simpler, because all Armv8 SoCs I've seen always use GIC (even the TI ones which used to use their own interrupt controller with v7a).. 

But I understand the negative implications it can have, i.e. the irq table creation needs this information. So if this info is unavailable at boot time, we will need to create the irq table at runtime (using dynamic memory), which is not a very good option.. May be others as well..